### PR TITLE
Move typescript definition file into the project

### PR DESCRIPTION
**Pull request by marduke182**
Move typescript definition from [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped) repository to the project.

This would help to prevent releases of **prosemirror-tables** before the type is updated in DefinitelyTyped repository.

This has been done in similar way for packages like **redux**.

### CODE REVIEW

**Comment by marduke182** [comment]
Maybe we could do something similar to what `redux` does too.

A deprecation warning about the old types.

`npm WARN deprecated @types/redux@3.6.0: This is a stub types definition for Redux (https://github.com/reactjs/redux). Redux provides its own type definitions, so you don't need @types/redux installed!`

**Comment by eshvedai** [comment]
@nathf can you please have a look?

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+  ranges: Array<SelectionRange<S>>;
+
+  map(doc: ProsemirrorNode<S>, mapping: any): any;
+  content(): Slice<S>;
+  replace(tr: Transaction<S>, content: Slice<S>): void;
+  replaceWith(tr: Transaction<S>, node: ProsemirrorNode<S>): void;
+  forEachCell(f: (node: ProsemirrorNode<S>, pos: number) => void): void;
+  isRowSelection(): boolean;
+  isColSelection(): boolean;
+  eq(other: any): boolean;
```
Comment (nathf) [inline index.d.ts]: can we use a more definitive type for `other`?

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+  anchor: number;
+  head: number;
+  $anchor: ResolvedPos<S>;
+  $head: ResolvedPos<S>;
+  $anchorCell: ResolvedPos<S>;
+  $headCell: ResolvedPos<S>;
+  empty: boolean;
+  ranges: Array<SelectionRange<S>>;
+
+  map(doc: ProsemirrorNode<S>, mapping: any): any;
```
Comment (nathf) [inline index.d.ts]: `mapping` can be `Mapping` type?

File: index.d.ts (hunk 1 above)
Comment (marduke182) [reply index.d.ts]: Sure, let me fix it. I just copy the file from DefinitelyTyped and I didn't check it 🙈

File: index.d.ts (hunk 2 above)
Comment (marduke182) [reply index.d.ts]: I used `Mappable`. That's the one that `Selection` type is using. Is this correct, @eshvedai ?

**Pull request merged**